Fix brew test by avoiding to use ENV.clear in formulas

jmechnich opened the pull request:
Following [advice](https://github.com/Homebrew/brew/issues/14752) by the homebrew core developers, clearing the shell environment is not a great idea for several reasons. In fact, it is not done in any of the other core formulas.
I have checked all gcc versions locally on macOS Ventura on ARM, and apparently only CPATH needs to be unset.

jmechnich commented:
Should I split this up in individual commits per formula? Or does it even have to be separated into multiple PRs?

ladislas commented:
let me review the workflows first, I haven't updated them in a while.

I'll let you know when you can rebase.

ladislas commented:
@jmechnich I've updated the CI

can you rebase and split this PR for each formula modified? 

it will also make things faster

jmechnich commented:
> @jmechnich I've updated the CI
> 
> can you rebase and split this PR for each formula modified?
> 
> it will also make things faster

just to clarify: split into individual PRs, or just the commits?

jmechnich committed:
avr-gcc@5: do not clear shell environment when running 'brew test'

jmechnich committed:
avr-gcc@8: do not clear shell environment when running 'brew test'

jmechnich committed:
avr-gcc@9: do not clear shell environment when running 'brew test'

jmechnich committed:
avr-gcc@10: do not clear shell environment when running 'brew test'

jmechnich committed:
avr-gcc@11: do not clear shell environment when running 'brew test'

jmechnich committed:
avr-gcc@12: do not clear shell environment when running 'brew test'

ladislas commented:
individual PRs

jmechnich commented:
Ok, maybe I should close this one first then :sweat_smile:
